### ISSUE TITLE
Hexadecimal input is parsed as number instead of integer

### ISSUE THREAD

**Issue by falkoschindler** [ianh/owl#28]
I was thinking about adding support for hexadecimal arguments in my grammar, when I noticed that OWL already parses things like "0xff". But why is every hex input parsed as a number and not an integer? In my application it messes with the type checking system, because some function might expect an integer and would refuse to accept "0xff".

For example, the grammar

```
input = argument{','}
argument =
  integer : int
  number : num
```

with the input

```
1, 1.2, 0xff, 0xff.f
```

yields:

```
1           , 1.2         , 0xff        , 0xff.f      
argument:int  argument:num  argument:num  argument:num
```

Would it be possible to distinguish hex integers and hex floating point numbers without writing a new tokenize function?

**Comment by ianh** [comment]
Ah, this is because `strtod` is used to parse numbers.  It accepts hexadecimal numbers, while owl's hand-written integer parser doesn't.  This is definitely an oversight.

**Comment by ianh** [comment]
This should be fixed in 92c1b65.

**Comment by falkoschindler** [comment]
Awesome, that was quick! It seems to be working nicely.
Thanks a lot!
